src/query.c

```diff
@@ -471,6 +471,14 @@ IndexIterator *Query_EvalTokenNode(QueryEvalCtx *q, QueryNode *qn) {
   // we can just use the optimized score index
   int isSingleWord = q->numTokens == 1 && q->opts->fieldmask == RS_FIELDMASK_ALL;
 
+  // If this is an empty value search, and the field does not index empty values
+  // throw an error
+  if (qn->tn.nen == NON_EXIST_EMPTY && !(FieldSpec_IndexesEmpty(IndexSpec_GetFieldByBit(q->sctx->spec, qn->opts.fieldMask)))) {
+    QueryError_SetErrorFmt(q->status, QUERY_ENOINDEX, "Field `%s` does not index empty values",
```

Comment:
Consider a positive message potentially assisting the user, such as "Field `%s` should enable `ISEMPTY` in the index SCHEMA in order to support empty values"